### ISSUE TITLE
Add PagerDuty auto resolution for Issue Alerts

### ISSUE THREAD

**Issue by cseas** [getsentry/sentry#55745]
### Problem Statement

Metric alerts have a way to auto-resolve them on PagerDuty automatically when the error volume goes down but the same isn't available for Issue Alerts. This creates a lot of extra work for our on-call engineers because they've to manually resolve each Issue Alert triggered by Sentry even though the issue's volume has gone down below the threshold it got triggered for.

### Solution Brainstorm

When the issue's volume goes below the threshold it got triggered for, it should get auto-resolved.

I gather from this comment https://github.com/getsentry/sentry/issues/34372#issuecomment-1121348864 that the suggestion is to use Metric Alerts instead but that doesn't work for our use-case. Metric alerts report a collection of issues so it gets hard to identify new high volume issues with them if existing high volume issues exist already. Issue Alerts help us solve that by triggering alerts for specific issues, but right now they're a lot of trouble for our on-call engineers since they never get auto-resolved.

### Product Area

Alerts

**Comment by rachrwang** [comment]
Hi @cseas - I'm Rachel, PM for the Issues product.

in Project -> General Settings, we support configuring "auto resolve" as the amount of time in which we stop receiving errors for a given issue, see:
![Screenshot 2023-09-06 at 7 50 45 AM](https://github.com/getsentry/sentry/assets/82916311/42399ff2-7043-4977-8a21-ae07b3d97306)

Would this address your use case, or is there some other definition that you'd like to use for "auto-resolve"? 

I'd also like to be better understand what you mean by "they're a lot of trouble for our on-call" -> is that related to receiving the notification, looking at the issue in the product, or something else? We're actively trying to improve the alerts/ notifications experience - and would appreciate any feedback you have for us!

**Comment by rachrwang** [comment]
@cseas - thank you for the detailed explanation, it's very helpful context. 

To what you wrote here: "So only for Sentry's Issue Alerts, we need to educate all our on-call engineers that these need to be resolved manually." -> to clarify, you're referring to manual resolution in PagerDuty? 

Would you be open to connecting live with me to chat more about your use case, especially on how you've configured your issue & metric alerts? You can schedule with me directly here: https://calendly.com/rachel-wang-sentry/30min?month=2023-09, or feel free to email me at rachel.wang@sentry.io. My team is working on projects related to alerts & notifications, and I'd like to learn more about your use case to help inform our upcoming roadmap.

**Comment by cseas** [comment]
> you're referring to manual resolution in PagerDuty?

Yes, the "time to resolve" of PagerDuty incidents is the only metric we track. We just use Sentry to funnel incidents into PagerDuty. If the issue only auto-resolves on Sentry but the related Issue Alert triggered by it remains on PagerDuty, that needs manual resolution and that's what we're looking for help on.

Happy to connect for more feedback and explain our setup. I've sent you a mail.

**Comment by jvineet** [comment]
@rachrwang Any update on where this feature request is at. We recently upgraded sentry to `v24.4.2` from `v23.3.1` and we find ourself in a similar predicament w.r.t sentry cron monitor alerts. The addition of failure and recovery tolerances in the cron monitor configuration has been a huge improvement, where sentry automatically creates a "monitor failure" issue after a successive # of cron failures and then automatically resolves the issue if there were a certain # of successful executions.

This works great from the cron observability perspective. We have set up Issue based alerts on failure events for some crons that send notifications to Slack and Pagerduty. But once the issue is resolved automatically by sentry, after there were subsequent successful cron executions, the related alerts to Slack and Pagerduty are not automatically resolved. This makes it somewhat cumbersome for the on-call engineers who need to go and resolve these alerts manually. 

It would be great if there was an option to set auto-resolution for Issue based alerts when the underlying issue gets resolved automatically.

For more context, we can't use metrics based alerting here, for two broad reasons.
1. It looks like the monitor failure issues belong to `issuePlatform` dataset and metrics-based alerts don't work on this dataset.
2. We have 100s of crons that have a wide variety of cron intervals. So metrics-base alerting is just not a good fit for setting up alerts on cron monitor failures.

**Comment by jvineet** [comment]
@rachrwang Just wanted to follow up on this problem, since it has been a whole year, to check if this is still something the sentry team is actively pursuing. Having auto-resolution in Pagerduty for Issue-based alerts, especially the ones with Issue category as `cron`, will be immensely helpful.

**Comment by sentaur-athena** [comment]
This issue is not particularly about pager duty so sending over to alerts team for a better look.

@jvineet issue alerts are very different from crons and metrics in nature as they group incoming errors. We do have the auto-resolve feature as Rachel mentioned above that you can use. In future you will also be able to define custom workflows for each alert.

**Comment by codingminions** [comment]
Is there any update here regarding the automatic resolution of PagerDuty incidents when the number of events for an issue falls below the trigger threshold? Or any similar 'below threshold' handling for PagerDuty based on issue alerts?

**Comment by souredoutlook** [comment]
Hey @codingminions the plan for [Metric Monitors ](https://github.com/getsentry/sentry/discussions/101960) is to allow folks to query for the number of events over a period of time grouped by issue... this would basically create pagerduty incidents that autoresolve per issue.

Metric Monitors are in closed beta currently and this feature would be slated to be developed after they are released. Feel free to send me an email at nicholas.meisenheimer@sentry.io if you'd like to be included in the closed Beta.

**Comment by codingminions** [comment]
Thanks @souredoutlook for sharing this. I just sent an email over with our org details and interest in the closed beta. Looking forward to it! Cheers 🚀  ...
